spectree/models.py

```diff
@@ -54,10 +58,38 @@ class SecureType(str, Enum):
     OPEN_ID_CONNECT = "openIdConnect"
 
 
-class SecuritySchemesData(BaseModel):
+class SecuritySchemeData(BaseModel):
+    """
+    Security scheme data
+    """
+
     type: SecureType = Field(..., description="Secure scheme type")
     name: str = None
     field_in: str = Field(None, alias="in")
     scheme: str = None
     openIdConnectUrl: str = None
     flows: dict = None
+
+    class Config:
+        validate_assignment = True
+
+
+class SecurityScheme(BaseModel):
+    """
+    Named security scheme
+    """
+
+    name: str = Field(
+        ...,
+        description="Custom security scheme name. Can only contain - [A-Za-z0-9-._]",
+    )
```

Comment:
What's the difference between `SecurityScheme.name` and `SecuritySchemeData.name`?

Reply:
> SecurityScheme.name

Custom name for auth scheme, init in `components->securitySchemes` and then can be used in `route->method->security`. Need validation.

> SecuritySchemeData.name

Custom variable name for header, for example. Validation not needed.

![Screenshot_2021-05-05_21-39-58](https://user-images.githubusercontent.com/16991365/117192858-de644d80-adea-11eb-9c41-4074ebe5033f.png)

Reply:
Add more descriptions